packages/file-service/src/modules/file/file.controller.ts

```diff
@@ -0,0 +1,86 @@
+import { Controller, Inject } from '@nestjs/common';
+import { FileProto } from 'juno-proto';
+import { ClientGrpc } from '@nestjs/microservices';
+import { status } from '@grpc/grpc-js';
+import { RpcException } from '@nestjs/microservices';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+const { FILE_DB_SERVICE_NAME } = FileProto;
+
+@Controller()
+@FileProto.FileServiceControllerMethods()
+export class FileController implements FileProto.FileServiceController {
+  private fileService: FileProto.FileDbServiceClient;
+
+  constructor(@Inject(FILE_DB_SERVICE_NAME) private fileClient: ClientGrpc) {}
+
+  onModuleInit() {
+    this.fileService =
+      this.fileClient.getService<FileProto.FileDbServiceClient>(
+        FILE_DB_SERVICE_NAME,
+      );
+  }
+
+  async downloadFile(
+    request: FileProto.DownloadFileRequest,
+  ): Promise<FileProto.DownloadFileResponse> {
+    console.log(request);
+    if (
+      !request ||
+      !request.fileName ||
+      !request.data ||
+      !request.bucket ||
+      !request.provider
+    ) {
+      throw new RpcException({
+        code: status.INVALID_ARGUMENT,
+        message: 'Must provide filename, data, bucket, and provider',
+      });
+    }
+
+    const metadata = request.data;
```

Comment:
Don't take request.data here, we don't need a `data` field in the Request, use the `provider` + `configId` to look up the provider in the DB service (CRUD methods from last sprint) and use the metadata field there to instantiate S3Client